### ISSUE TITLE
Multiple AZ-hop deployments

### ISSUE THREAD

**Issue by matt-chan** [Azure/az-hop#972]
Hi Xavier,

Thanks for creating az-hop.

Is there a way of managing multiple AZ-hop deployments from within a single clone of the repo? Right now it seems like it leaves a bunch of stateful things, like the tfstate file, the .ok file, and the options.json for packer. 

It would be nice if there was a way to switch between different deployments (or wipe it back to default). I've been running into an issue where I try to create a new deployment in the config (by specifying a new resource group) but TF tries to migrate things to the new resource group instead and then a bunch of things break. 

Thanks!
Matt

**Comment by matt-chan** [comment]
@ltalirz

**Comment by matt-chan** [comment]
Also, I just noticed the documentation about azhop-state.sh. I will try to write a script that wraps it in more of a entry-exit fashion...

**Comment by ltalirz** [comment]
related to https://github.com/Azure/az-hop/issues/948

**Comment by xpillons** [comment]
you are right this is an issue. manually removing the tfstate can solve this. A nicer solution would be to have dedicated directories for each deployments. and as you said using the azhop-state.sh will allow you to save state files in a container and download it later of you need to update a deployment. 
What out customers are doing is to use a deployer VM which will have a clone or a fork of the repo plus config and states of a deployment, and have one deployer VM per environment being built, but this may not be a good scenario if you have plenty of environments to manage.
